python/pyarrow/table.pxi

```diff
@@ -5078,11 +5082,24 @@ def concat_tables(tables, c_bool promote=False, MemoryPool memory_pool=None):
         CConcatenateTablesOptions options = (
             CConcatenateTablesOptions.Defaults())
 
+    if "promote" in kwargs:
+        warnings.warn(
+            "promote has been superseded by mode='default'.", FutureWarning)
+        if kwargs['promote'] is True:
+            promote_options = "permissive"
```

Comment:
```suggestion
            promote_options = "default"
```

Isn't that needed to preserve the current default behaviour?

Reply:
I thought we agreed on promoting when `promote=True`, but this also works for me.